clientlibs/js/src/functions/getExperimentCondition.ts

```diff
@@ -52,25 +76,51 @@ export class Assignment {
       return null;
     }
   }
+
+  public markDecisionPoint(status: MARKED_DECISION_POINT_STATUS, uniquifier?: string, clientError?: string) {
+    return markExperimentPoint(
+      this._markObject.url,
+      this._markObject.userId,
+      this._markObject.token,
+      this._markObject.clientSessionId,
+      this._site,
+      this._target,
+      this._conditionCode,
+      status,
+      this._markObject.getAllExperimentData,
+      uniquifier,
+      clientError
+    );
+  }
 }
 
 /**
  * @hidden
  */
 export default function getExperimentCondition(
-  experimentConditionData: IExperimentAssignmentv4[],
+  experimentConditionData: IExperimentAssignmentv5[],
   site: string,
-  target: string
+  target: string,
+  markObject: markObejectType
 ): Assignment {
   if (experimentConditionData) {
     const result = experimentConditionData.find((data) => data.target === target && data.site === site);
 
     if (result) {
       const assignment = new Assignment(
-        result.assignedCondition.conditionCode,
-        result.assignedCondition.payload,
-        result.assignedFactor
+        result.site,
+        result.target,
+        markObject,
+        result.assignedCondition[0].conditionCode,
+        result.assignedCondition[0].payload,
+        result.assignedFactor ? result.assignedFactor[0] : null
       );
+
+      // rotate the condition queue
+      result.assignedCondition.push(result.assignedCondition.shift());
```

Comment:
isn't this rotation supposed to happen after doing `markExperimentPoint`, not `getAssignment`? The Java library is doing this on mark.

Reply:
Solved this part, now conditions will rotate on getAssignement call